sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosDiagnosticsTest.java

```diff
@@ -312,6 +317,26 @@ public void directDiagnostics() throws Exception {
             validateRegionContacted(createResponse.getDiagnostics(), testDirectClient.asyncClient());
             isValidJSON(diagnostics);
 
+            // read item to validate response and request session tokens are equal
+            String sessionToken = createResponse.getSessionToken();
+            CosmosItemResponse<InternalObjectNode> readResponse =
+                cosmosContainer.readItem(BridgeInternal.getProperties(createResponse).getId(),
+                    new PartitionKey(BridgeInternal.getProperties(createResponse).getId()),
+                    InternalObjectNode.class);
+            diagnostics = readResponse.getDiagnostics().toString();
+            assertThat(diagnostics).contains(String.format("\"requestSessionToken\":\"%s\"", sessionToken));
+
+            // use batch operation to check that user-set session token is being passed in
```

Comment:
can we also have a case for normal create item operation to show that the requestSessionToken will not be sent